File: app/Models/Product.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class Product extends Model
{
    use HasFactory, SoftDeletes;

    /**
     * The attributes that are mass assignable.
     *
     * @var list<string>
     */
    protected $fillable = [
        'nom_commercial',
        'description',
        'composant_actif',
        'dosage_recommande',
        'delai_avant_recolte',
        'type',
        'avantages',
        'usage_method',
        'safety_instructions',
        'image',
    ];

    /**
     * Get the attributes that should be cast.
     *
     * @return array<string, string>
     */
    protected function casts(): array
    {
        return [
            'type' => \App\Enums\ProductTypeEnum::class,
            'delai_avant_recolte' => 'integer',
        ];
    }

    /**
     * Get the cultures for this product.
     */
    public function cultures()
    {
        return $this->belongsToMany(Culture::class, 'culture_product')
            ->withPivot('dosage_specifique', 'notes')
            ->withTimestamps();
    }
}

File: mobile/lib/src/models/product.dart
import 'package:equatable/equatable.dart';

/// Mirrors App\Http\Resources\ProductResource.
class Product extends Equatable {
  const Product({
    required this.id,
    required this.nomCommercial,
    this.description,
    this.composantActif,
    this.dosageRecommande,
    this.delaiAvantRecolte,
    this.type,
    this.avantages,
    this.usageMethod,
    this.safetyInstructions,
    this.image,
    this.pivotDosage,
    this.pivotNotes,
  });

  final int id;
  final String nomCommercial;
  final String? description;
  final String? composantActif;
  final String? dosageRecommande;
  final int? delaiAvantRecolte;
  final String? type;
  final String? avantages;
  final String? usageMethod;
  final String? safetyInstructions;
  final String? image;

  /// From the culture<->product pivot when present.
  final String? pivotDosage;
  final String? pivotNotes;

  factory Product.fromJson(Map<String, dynamic> json) {
    final pivot = json['pivot'] as Map<String, dynamic>?;
    return Product(
      id: json['id'] as int,
      nomCommercial: json['nom_commercial'] as String? ?? '',
      description: json['description'] as String?,
      composantActif: json['composant_actif'] as String?,
      dosageRecommande: json['dosage_recommande'] as String?,
      delaiAvantRecolte: (json['delai_avant_recolte'] as num?)?.toInt(),
      type: json['type'] as String?,
      avantages: json['avantages'] as String?,
      usageMethod: json['usage_method'] as String?,
      safetyInstructions: json['safety_instructions'] as String?,
      image: json['image'] as String?,
      pivotDosage: pivot?['dosage_specifique'] as String?,
      pivotNotes: pivot?['notes'] as String?,
    );
  }

  @override
  List<Object?> get props => [id, nomCommercial, type];
}
